### PR TITLE
Feature/trivial gate inversions

### PR DESCRIPTION
**Pull request by nbronn**
### Summary

The reversal of a static subset of gates is known when the transpiler needs to flip the direction. This PR adds the knowledge that symmetric gates (`rxx`, `ryy`, `rzz`) are their own flipped gate to `gate_direction.py`.

### CODE REVIEW

**Review by Cryoris — COMMENTED** [review]
The CZ gate, which is also symmetric, is handled a bit differently here  where the qubit arguments are explicitly swapped (see L84). Do we need to do this here too? If not, we could also avoid applying the CZ gate and just skip it like you suggested here 🤔

**Comment by nbronn** [comment]
That's a great point @Cryoris, I tried implementing `cz` in the same way and failed the second assertion of the unit test `test.python.transpiler.test_gate_direction.TestGateDirection.test_target_static` it seems because it wants the order of the qubits reversed in the DAG. If this is indeed necessary for `cz` then it should apply to `rxx`, `ryy`, and `rzz` as well.

File: qiskit/transpiler/passes/utils/gate_direction.py [hunk 1]
```diff
@@ -132,6 +166,14 @@ def _run_coupling_map(self, dag, wire_map, edges=None):
...
                 elif node.name == "rzx":
                     dag.substitute_node_with_dag(node, self._rzx_dag(*node.op.params))
+                elif node.name == "rxx":
+                    dag.substitute_node_with_dag(node, self._rxx_dag(*node.op.params))
+                elif node.name == "ryy":
+                    dag.substitute_node_with_dag(node, self._ryy_dag(*node.op.params))
+                elif node.name == "rzz":
+                    dag.substitute_node_with_dag(node, self._rzz_dag(*node.op.params))
+                # elif node.name in ["cz", "rxx", "ryy", "rzz"]:
+                #     continue
```
Comment (1ucian0) [inline qiskit/transpiler/passes/utils/gate_direction.py:176]: it this needed?

Comment (nbronn) [reply qiskit/transpiler/passes/utils/gate_direction.py]: Whoops... removed!

**Comment by 1ucian0** [comment]
Thanks a lot for working on this Nick!

Can you add a `fixes` release note?

File: releasenotes/notes/fix-trivial-gate-inversions-1e39293d59bc6027.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+fixes:
+  - |
+    Added gate reversal directions for the symmetric rxx, ryy, and rzz gates.
```
Comment (1ucian0) [inline releasenotes/notes/fix-trivial-gate-inversions-1e39293d59bc6027.yaml:4]:
```suggestion
    Added gate reversal directions for the symmetric :class:`~RXXGate`, :class:`~RYYGate`, and :class:`~RZZGate` gates.
```

**Comment by 1ucian0** [comment]
I extended the coverage with [c1808ed](https://github.com/Qiskit/qiskit-terra/pull/9262/commits/c1808ed03495ac7075f9e1eaf9b6bb2f02119ea7)

**Review by 1ucian0 — APPROVED** [review]
Thanks for the PR!

**Pull request merged**